CMakeLists.txt

```diff
@@ -267,9 +268,13 @@ set(CMAKE_C_FLAGS_DEBUGOPT "")
 
 target_compile_options(${PROJECT_NAME} PRIVATE -pedantic -std=gnu99 -Wall -Wimplicit -Wunused -Wcomment -Wchar-subscripts
         -Wuninitialized -Wshadow -Wcast-align -Wwrite-strings -Wno-deprecated-declarations -Wno-unknown-pragmas -Wformat-security
-        -Wno-missing-braces -Wsign-compare -Wno-strict-prototypes -Wa,--noexecstack
+        -Wno-missing-braces -Wno-strict-prototypes -Wa,--noexecstack
 )
 
+if (S2N_STRICT_CHECK)
+    target_compile_options(${PROJECT_NAME} PRIVATE -Wsign-compare -DS2N_STRICT_CHECK)
+endif()
```

Comment:
So now we only have "PRIVATE". Do you know why PUBLIC isn't required?

Reply:
Based on reading about CMake Inhertiance and discussion with Doug, all the options following `PUBLIC` will be used by current target and other targets that have dependencies on current target. In this case, scoping comes into picture when we're integrated within larger projects. If scoping was wrong, the `crt` tests would have failed. The tests passing makes sense because `crt` isn’t going to depend on cmake flags.  `PRIVATE` will only link to current target.